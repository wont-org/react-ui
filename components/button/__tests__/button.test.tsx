import React from 'react'
import {
    create,
    // act,
} from 'react-test-renderer'
import Button from '../index'
import mountTest from '../../../tests/shared/mountTest'


describe('Button test', () => {
    mountTest(() => <Button label="button" />)
    it('label', () => {
        const testButtonLabel = create(
            <Button label="button" />,
        )
        expect(testButtonLabel).toMatchSnapshot()
    })
    it('type', () => {
        const testButtonType = create(
            <Button label="button" type="primary" />,
        )
        expect(testButtonType).toMatchSnapshot()
    })
    it('size', () => {
        const testButtonType = create(
            <Button label="button" type="primary" size="medium" />,
        )
        expect(testButtonType).toMatchSnapshot()
    })
})
