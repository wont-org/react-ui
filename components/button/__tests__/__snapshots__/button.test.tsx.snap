// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test label 1`] = `
<button
  className="wont-btn wont-btn-primary wont-btn-medium"
  type="button"
>
  button
</button>
`;

exports[`Button test size 1`] = `
<button
  className="wont-btn wont-btn-primary wont-btn-medium"
  type="button"
>
  button
</button>
`;

exports[`Button test type 1`] = `
<button
  className="wont-btn wont-btn-primary wont-btn-medium"
  type="button"
>
  button
</button>
`;
